install.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
+
+set -e
+
+temp_dir=$(mktemp -d /tmp/xh.XXXXXXXX)
+cd "$temp_dir"
+
+if [ "$(uname -s)" = "Darwin" ] && [ "$(uname -m)" = "x86_64" ]; then
+    target="x86_64-apple-darwin"
+elif [ "$(uname -s)" = "Linux" ] && [ "$(uname -m)" = "x86_64" ]; then
+    target="x86_64-unknown-linux-musl"
+elif [ "$(uname -s)" = "Linux" ] && [ "$(uname -m)" = "amd64" ]; then
+    target="x86_64-unknown-linux-musl"
+elif [ "$(uname -s)" = "Linux" ] && ( uname -m | grep -q "arm" ); then
+    target="arm-unknown-linux-gnueabihf"
+else
+    echo "Unsupported OS or architecture"
+    exit 1
+fi
+
+if which curl > /dev/null; then
+    fetch='curl -sSL -o'
+elif which wget > /dev/null; then
+    fetch='wget -nv -O'
+else
+    echo "Can't find curl or wget, can't download package"
+    exit 1
+fi
+
+echo "Detected target: $target"
+
+url=$(
+    $fetch - https://api.github.com/repos/ducaale/xh/releases/latest |
+    grep -wo -m1 "https://.*$target.tar.gz" || true
+)
```

Comment:
On Raspberry Pi 4, this emits the warning `(23) Failed writing body`. I am guessing it is related to this issue https://stackoverflow.com/questions/16703647/why-does-curl-return-error-23-failed-writing-body

Reply:
Fixed